packages/pg-connection-string/test/toClientConfig.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const chai = require('chai')
+const expect = chai.expect
+chai.should()
+
+const { parse, toClientConfig } = require('../')
+
+describe('toClientConfig', function () {
+  it('converts connection info', function () {
+    const config = parse('postgres://brian:pw@boom:381/lala')
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.user.should.equal('brian')
+    clientConfig.password.should.equal('pw')
+    clientConfig.host.should.equal('boom')
+    clientConfig.port.should.equal(381)
+    clientConfig.database.should.equal('lala')
+  })
+
+  it('converts query params', function () {
+    const config = parse(
+      'postgres:///?application_name=TheApp&fallback_application_name=TheAppFallback&client_encoding=utf8&options=-c geqo=off'
+    )
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.application_name.should.equal('TheApp')
+    clientConfig.fallback_application_name.should.equal('TheAppFallback')
+    clientConfig.client_encoding.should.equal('utf8')
+    clientConfig.options.should.equal('-c geqo=off')
+  })
+
+  it('converts SSL boolean', function () {
+    const config = parse('pg:///?ssl=true')
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.ssl.should.equal(true)
+  })
+
+  it('converts sslmode=disable', function () {
+    const config = parse('pg:///?sslmode=disable')
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.ssl.should.equal(false)
+  })
+
+  it('converts sslmode=noverify', function () {
+    const config = parse('pg:///?sslmode=no-verify')
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.ssl.rejectUnauthorized.should.equal(false)
+  })
+
+  it('converts other sslmode options', function () {
+    const config = parse('pg:///?sslmode=verify-ca')
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.ssl.should.deep.equal({})
+  })
+
+  it('converts other sslmode options', function () {
+    const config = parse('pg:///?sslmode=verify-ca')
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.ssl.should.deep.equal({})
+  })
+
+  it('converts ssl cert options', function () {
+    const connectionString =
+      'pg:///?sslcert=' +
+      __dirname +
+      '/example.cert&sslkey=' +
+      __dirname +
+      '/example.key&sslrootcert=' +
+      __dirname +
+      '/example.ca'
+    const config = parse(connectionString)
+    const clientConfig = toClientConfig(config)
+
+    clientConfig.ssl.should.deep.equal({
+      ca: 'example ca\n',
+      cert: 'example cert\n',
+      key: 'example key\n',
+    })
+  })
+
+  it('converts unix domain sockets', function () {
+    const config = parse('socket:/some path/?db=my[db]&encoding=utf8&client_encoding=bogus')
+    const clientConfig = toClientConfig(config)
+    clientConfig.host.should.equal('/some path/')
+    clientConfig.database.should.equal('my[db]', 'must to be escaped and unescaped through "my%5Bdb%5D"')
```

Comment:
This feels like it should've thrown an error?

Reply:
this is current behavior according to: https://github.com/brianc/node-postgres/blob/df0f4d19fbf0349c6b41e8982e058f7377ee3bea/packages/pg-connection-string/test/parse.js#L43